frontend/lib/resolution/InitResolver.cpp

```diff
@@ -821,14 +821,24 @@ bool InitResolver::handleAssignmentToField(const OpCall* node) {
 
     // TODO: Anything to do if the opposite is true?
     if (!isAlreadyInitialized) {
+      // Recompute field type in case it depends on a recently-instantiated
+      // field. For example, ``var curField : typeField;``.
+      auto rf = resolveFieldDecl(ctx_, currentRecvType_->getCompositeType(), fieldId, DefaultsPolicy::IGNORE_DEFAULTS);
+      auto initialFieldType = rf.fieldType(0);
+
```

Comment:
This is not going to work in the presence of multi-declarations. See: 

https://github.com/chapel-lang/chapel/blob/9297b7ed36bc6ebcefd58434b8843ed432c4e41c/frontend/lib/resolution/resolution-queries.cpp#L810-L820

To ensure that multi-decls are properly resolved, `resolveFieldDecl` can end up returning multiple things -- see this piece of `helpSetFieldTypes`:

https://github.com/chapel-lang/chapel/blob/9297b7ed36bc6ebcefd58434b8843ed432c4e41c/frontend/lib/resolution/resolution-queries.cpp#L730-L732

So, you can't use `rf.fieldType(0)`. Rather, I think you have to iterate the elements in `initialFieldType` and find the ID you were looking for. That, or adjust `resolveFieldDecl` + `initialFieldType` to only add the field in question.